jetstream_pt/ray_engine.py

```diff
@@ -38,6 +39,8 @@ def __init__(
     self.batch_size = batch_size
     self.is_disaggregated = is_disaggregated
     self.pod_slice_name = pod_slice_name
+    if not self.is_disaggregated:
+      self._lock = threading.Lock()
```

Comment:
lets use RLock. For lock if the thread already have the lock and it try to lock again it will deadlock

Reply:
Use lock (without R) should be the right way. The jetstream may have the issue before (same thread call prefill or decode concurrently), there is no recursion in prefill or decode code.  

If a thread call the both prefill and decode  or itself method multiple time without unlock, then there is a bug on jetstream.